src/app/new-contact/formValidators.ts

```diff
@@ -0,0 +1,23 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+export const noSpacesValidator: ValidatorFn = (
+  control: AbstractControl
+): ValidationErrors | null => {
+  if ((control.value as string).indexOf(' ') !== -1) {

```

Comment:
Just be aware that `as string` doesn't cast the variable to a string - just tells typescript to act like it is a string. If you used this validator on a `<input type="number" />` it would cause an error